crates/paralegal-policy/src/context.rs

```diff
@@ -308,15 +456,20 @@ impl Context {
             .0;
 
         while let Some(current) = queue.pop() {
-            for sink in flow.get(&current).into_iter().flatten() {
-                if is_checkpoint(sink) {
+            let Some(datasource) = current.node.as_data_source() else {
+				continue
+			};
```

Comment:
Since our `is_terminal` and `is_checkpoint` functions now accept any `Node` I think we need to add check blocks similar to line 467 and following here.
It doesn't need to check `seen` (because that's just for the DFS internally) but it should check if we've hit a checkpoint or terminal

Reply:
Sure, I moved the check blocks from line 467 up so as to not double count.